src/core/player.ts

```diff
@@ -477,6 +477,15 @@ export default class Player {
 		this.methodSeekTo(newTime)
 	}
 
+	/**
+	 * Set the media source
+	 * @param source New media source URL
+	 */
+	setSource(source: string) {
```

Comment:
I'm not sure if the method should be here, because the Player class is used by all providers with inheritance (Youtube, Dailymotion, HTML5, etc.) and this is a feature only for HTML5 video and audio. This is the first time this case has arisen I believe.

This must also be documented. I will check

Reply:
Maybe can help:
https://github.com/videojs/video.js/blob/8842d376ed3d834ad3ce0a5e662fe446513ea0bd/src/js/tech/html5.js#L1932-L1942
Related HTML5 doc: https://html.spec.whatwg.org/multipage/media.html#dom-media-src

Reply:
Video.js method is `setSource`

Reply:
Maybe, if this is a feature only for HTML, it can be not added but the doc should be updated with related use cases on how can be done.
I wrote below my use case with HTML and where I have some problem.

Reply:
> If the method is for only one provider, it should be is the dedicated provider (`html5`)

Agree.

Can the providers be used as modular plugin? (new feature)

Reply:
I did a commit to move `setSource` method from player core to html5 provider.